hooks/pre/30-el8_upgrade_postgresql.rb

```diff
@@ -0,0 +1,32 @@
+def postgresql_upgrade(new_version)
+  log_and_say(:info, "Upgrading PostgreSQL to #{new_version}")
```

Comment:
We could just do a log of this at the notice level, then instead of 

```
2024-03-06 17:47:42 [NOTICE] [root] Loading installer configuration. This will take some time.
2024-03-06 17:47:44 [NOTICE] [root] Running installer with log based terminal output at level NOTICE.
2024-03-06 17:47:44 [NOTICE] [root] Use -l to set the terminal output log level to ERROR, WARN, NOTICE, INFO, or DEBUG. See --full-help for definitions.
Upgrading PostgreSQL to 13
```

We'd get

```
2024-03-06 17:47:42 [NOTICE] [root] Loading installer configuration. This will take some time.
2024-03-06 17:47:44 [NOTICE] [root] Running installer with log based terminal output at level NOTICE.
2024-03-06 17:47:44 [NOTICE] [root] Use -l to set the terminal output log level to ERROR, WARN, NOTICE, INFO, or DEBUG. See --full-help for definitions.
2024-03-06 17:47:44 [NOTICE] [root] Upgrading PostgreSQL to 13
```

Reply:
I am also noticing that it just sorta sits after the above message with no indicator that other things have happened by default, and so it appears like it's just hung. Our default level is `:notice` but the execute commands default to `:debug`. I think it would be useful here to log a few of the other actions as `:notice` to give a sense of things happening.

Reply:
Timing info from my run:

- 1 minute to stop services
- 2 minutes to run `dnf module switch-to postgresql:13 -y`
- 6 minutes for... package ensure?

```
2024-03-06 17:50:56 [INFO  ] [pre] Ensuring postgresql, postgresql-server, postgresql-upgrade, postgresql-contrib to package state latest
2024-03-06 17:56:34 [DEBUG ] [root] Executing: sed -i '/^data_directory/d' /var/lib/pgsql/data/postgresql.conf
```

- 15 seconds to run the postgres --upgrade command

Reply:
That package ensure probably contains selinux stuff

Reply:
Otoh, switch-to already does upgrade most packages, that ensure is just to ensure (and add the upgrade pkg)

Reply:
```
[root@centos8-stream-katello-nightly ~]# foreman-installer 
2024-03-07 08:48:20 [NOTICE] [root] Loading installer configuration. This will take some time.
2024-03-07 08:48:23 [NOTICE] [root] Running installer with log based terminal output at level NOTICE.
2024-03-07 08:48:23 [NOTICE] [root] Use -l to set the terminal output log level to ERROR, WARN, NOTICE, INFO, or DEBUG. See --full-help for definitions.
2024-03-07 08:48:27 [NOTICE] [pre] Performing upgrade of PostgreSQL to 13
2024-03-07 08:49:29 [NOTICE] [pre] Upgrading PostgreSQL packages
2024-03-07 08:49:42 [NOTICE] [pre] Migrating PostgreSQL data
2024-03-07 08:49:46 [NOTICE] [pre] Analyzing the new PostgreSQL cluster
2024-03-07 08:49:47 [NOTICE] [pre] Upgrade to PostgreSQL 13 completed
2024-03-07 08:49:50 [NOTICE] [configure] Starting system configuration.
```